jujugui/static/gui/src/app/components/generic-input/generic-input.js

```diff
@@ -119,9 +120,12 @@ YUI.add('generic-input', function() {
       Handle blur events for the input.
       @method _blurHandler
     */
-    _blurHandler: function() {
+    _blurHandler: function(e) {
       this.setState({focus: false});
       this.validate();
+      if (this.props.onBlur && typeof this.props.onBlur === 'function') {
```

Comment:
Does this need to check for the function type? I think you'll get a prop type error from `onBlur: React.PropTypes.func` above.